### PR TITLE
feat!: remove all code related to arithmetic rewriting

### PR DESCRIPTION
**Pull request by kalzoo**
Clean removal of all arithmetic rewriting code.

Breaking change, of course; several public modules, structs, and functions removed from both rust and python packages.

### CODE REVIEW

File: crates/lib/examples/execute.rs [hunk 1]
```diff
@@ -19,39 +22,47 @@ async fn quilc_client() -> rpcq::Client {
 
 #[tokio::main]
 async fn main() {
-    let mut exe = Executable::from_quil(PROGRAM).with_quilc_client(Some(quilc_client().await));
+    let mut exe = Executable::from_quil(PROGRAM); //.with_quilc_client(Some(quilc_client().await));
```
Comment (Shadow53) [inline crates/lib/examples/execute.rs:25]: Commented out code, should probably be restored

File: crates/lib/examples/execute.rs [hunk 2]
```diff
@@ -19,39 +22,47 @@ async fn quilc_client() -> rpcq::Client {
...
-    dbg!(&result);
+    // let result = exe.retrieve_results(handle.clone()).await;
+    // if !cancelled {
+    //     println!("Job was not cancelled");
+    //     assert!(result.is_ok());
+    // } else {
+    //     println!("Job was cancelled");
+    //     assert!(result.is_err());
+    // }
+    // dbg!(&result);
```
Comment (Shadow53) [inline crates/lib/examples/execute.rs:67]: Should all of this remain commented too?

**Review by Shadow53 — COMMENTED** [review]
Changes need to be cleaned up in `crates/lib/examples/execute.rs`, but the module removal itself looks fine.

CI checks failed, btw.

File: crates/lib/examples/execute.rs (hunk 2 above)
Comment (kalzoo) [reply crates/lib/examples/execute.rs]: whoops, nope, removed

**Review by jselig-rigetti — APPROVED** [review]

**Pull request merged**